modules/serverless-logic-security-example-oauth-authentication.adoc

```diff
@@ -0,0 +1,102 @@
+// Module included in the following assemblies:
+// serverles-logic/serverless-logic-authentication-openapi-services
+
+:_mod-docs-content-type: REFERENCE
+[id="serverless-logic-security-example-oauth-authentication_{context}"]
+= Example of OAuth 2.0 authentication
+
+The following example shows how to secure an OpenAPI operation using the OAuth 2.0 `clientCredentials` flow. The OpenAPI specification defines the `sayHelloOauth2` operation, which uses the `oauth-example` security scheme. Unlike simpler authentication methods such as HTTP Basic or API keys, OAuth 2.0 authentication requires additional integration with the Quarkus OpenID Connect (OIDC) Client.
+
+.Example OpenAPI specification with OAuth 2.0
+[source,json]
+----
+{
+  "openapi": "3.1.0",
+  "info": {
+    "title": "Oauth2 Scheme Example",
+    "version": "1.0"
+  },
+  "paths": {
+    "/hello-with-oauth2": {
+      "get": {
+        "operationId": "sayHelloOauth2",
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "text/plain": {
+                "schema": {
+                  "type": "string"
+                }
+              }
+            }
+          }
+        },
+        "security": [
+          {
+            "oauth-example": []
+          }
+        ]
+      }
+    }
+  },
+  "components": {
+    "securitySchemes": {
+      "oauth-example": {
+        "type": "oauth2",
+        "flows": {
+          "clientCredentials": {
+            "authorizationUrl": "https://example.com/oauth",
+            "tokenUrl": "https://example.com/oauth/token",
+            "scopes": {}
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+In this example, the `sayHelloOauth2` operation is protected by the `oauth-example` security scheme, which uses the `clientCredentials` flow for token-based authentication.
+
+[id="serverless-logic-security-oauth-support-oidc-client-filter-extention_{context}"]
+== OAuth 2.0 Support with the OIDC Client filter extension
+
+OAuth 2.0 token management is handled by a Quarkus `OidcClient`. To enable this integration, you must add the Quarkus OIDC Client Filter extension to your project as shown in the following examples:
+
+.Example of adding extension using Maven
+[source,xml]
+----
+<dependency>
+  <groupId>io.quarkus</groupId>
+  <artifactId>quarkus-oidc-client-filter</artifactId>
```

Comment:
@gmunozfe @wmedvede I think we need to reassess this doc with https://docs.quarkiverse.io/quarkus-openapi-generator/dev/client.html#_oauth2_authentication

I'm unsure which version of the openapi-generator we are using on 1.36.